portfolio/src/main/webapp/script.js

```diff
@@ -27,3 +27,50 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+/** Passes the response into handleResponse() once request is complete */
+function getGreeting() {
+  const responsePromise = fetch('/data');
+  responsePromise.then(handleResponse);
+}
+/** Calls for  */
+function deleteAllComments() {
+  fetch('/delete-data', {method: 'POST'} ).then(window.location.reload());
+}
+/** Converts response to text and passes the result into addGreetingToDom */
+function handleResponse(response) {
+  const textPromise = response.text();
+  textPromise.then(addGreetingToDom);
+}
+
+/** Adds a random quote to the DOM. */
+function addGreetingToDom(greeting) {
```

Comment:
Note that JSDoc includes params: https://google.github.io/styleguide/jsguide.html#jsdoc